## RNAseq notebook 3.2: SAM files and read counting


**Exercises**   
1) How many reads mapped to chromosome 20?  
2) Find the 76th read in the SAM file. Where did it map in the human genome? Now use blastn to map the read. Do the results agree with each other?  
3) Inspect the reference genome details in the SAM header. Beyond chromosomes, what else is included in the reference?

#### Exercise 1

In [3]:
%%bash
### Exercise 1
cd /mnt/c/Users/Jerry/Downloads/hisat_out/
awk '{if ($3 == 20) {print}}' SRR5454079.sam | wc -l

464329


Approx. 0.46 M alignments to chromosome 20. Is that expected by chance?  
Size of human chromosome 20 ... ca. 63 Mbp  
Size of human genome        ... ca. 3,235 Mbp  
Fraction of genomic DNA sequence represented by chr20 ... ca. 1.95%  

Number of alignments from SRR5454079: ca. 24 M  
Fraction of alignments attributed to chr20 ... ca. 1.92%  
  
Conclusion: chromosome 20 is not unusually transcriptionally active in this sample.

#### Exercise 2

In [5]:
%%bash
### Exercise 2
cd /mnt/c/Users/Jerry/Downloads/hisat_out/
grep SRR5454079.76 SRR5454079.sam | head -n 1

SRR5454079.76	16	20	327964	60	50M	*	0	0	ATACAGCGGGAAAAACTGAGGCACTTTGGTGCTAGGGGTTTGGGACTGAN	JJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFAA#	AS:i:-2	ZS:i:-2	XN:i:0	XM:i:1	XO:i:0	XG:i:0	NM:i:1	MD:Z:49G0	YT:Z:UU	NH:i:1



#### BLAST result for read SRR5454079.76 (obtained using the web interface): 
```
**Homo sapiens SRY-box 12 (SOX12), mRNA**
Sequence ID: NM_006943.3Length: 4658Number of Matches: 1
Related Information
Gene-associated gene details
UniGene-clustered expressed sequence tags
GEO Profiles-microarray expression data
NewGenome Data Viewer-aligned genomic context
    Range 1: 2394 to 2442GenBankGraphics
Next Match
Previous Match
Alignment statistics for match #1 Score	Expect	Identities	Gaps	Strand
91.6 bits(49) 	2e-17 	49/49(100%) 	0/49(0%) 	Plus/Plus

Query  1     ATACAGCGGGAAAAACTGAGGCACTTTGGTGCTAGGGGTTTGGGACTGA  49
             |||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  2394  ATACAGCGGGAAAAACTGAGGCACTTTGGTGCTAGGGGTTTGGGACTGA  2442

**Homo sapiens chromosome 20, GRCh38.p12 Primary Assembly**
Sequence ID: NC_000020.11Length: 64444167Number of Matches: 1
Related Information
PubChem BioAssay-bioactivity screening
NewGenome Data Viewer-aligned genomic context
Range 1: 327964 to 328012GenBankGraphics
Next Match
Previous Match
Alignment statistics for match #1 Score	Expect	Identities	Gaps	Strand
91.6 bits(49) 	2e-17 	49/49(100%) 	0/49(0%) 	Plus/Plus
Features:
1092 bp at 5' side: transcription factor SOX-12
18656 bp at 3' side: uncharacterized protein LOC101929937

Query  1       ATACAGCGGGAAAAACTGAGGCACTTTGGTGCTAGGGGTTTGGGACTGA  49
               |||||||||||||||||||||||||||||||||||||||||||||||||
Sbjct  327964  ATACAGCGGGAAAAACTGAGGCACTTTGGTGCTAGGGGTTTGGGACTGA  328012

```



From the HISAT2-aligned SAM file and the BLAST-aligned read we see the same genomic reference was found (chromosome 20, bases 327964 to 328012) which corresponds to a protein coding gene *SOX12*.

In [8]:
%%bash
#### Exercise 3
cd /mnt/c/Users/Jerry/Downloads/
head -n 50 ./hisat_out/SRR5454079.sam

@HD	VN:1.0	SO:unsorted
@SQ	SN:1	LN:248956422
@SQ	SN:10	LN:133797422
@SQ	SN:11	LN:135086622
@SQ	SN:12	LN:133275309
@SQ	SN:13	LN:114364328
@SQ	SN:14	LN:107043718
@SQ	SN:15	LN:101991189
@SQ	SN:16	LN:90338345
@SQ	SN:17	LN:83257441
@SQ	SN:18	LN:80373285
@SQ	SN:19	LN:58617616
@SQ	SN:2	LN:242193529
@SQ	SN:20	LN:64444167
@SQ	SN:21	LN:46709983
@SQ	SN:22	LN:50818468
@SQ	SN:3	LN:198295559
@SQ	SN:4	LN:190214555
@SQ	SN:5	LN:181538259
@SQ	SN:6	LN:170805979
@SQ	SN:7	LN:159345973
@SQ	SN:8	LN:145138636
@SQ	SN:9	LN:138394717
@SQ	SN:MT	LN:16569
@SQ	SN:X	LN:156040895
@SQ	SN:Y	LN:57227415
@SQ	SN:KI270728.1	LN:1872759
@SQ	SN:KI270727.1	LN:448248
@SQ	SN:KI270442.1	LN:392061
@SQ	SN:KI270729.1	LN:280839
@SQ	SN:GL000225.1	LN:211173
@SQ	SN:KI270743.1	LN:210658
@SQ	SN:GL000008.2	LN:209709
@SQ	SN:GL000009.2	LN:201709
@SQ	SN:KI270747.1	LN:198735
@SQ	SN:KI270722.1	LN:194050
@SQ	SN:GL000194.1	LN:191469
@SQ	SN:KI270742.1	LN:186739
@SQ	SN:GL000205.2	LN:185591
@SQ	SN:GL000195.1	LN:182896
@SQ	SN:KI270736.1	LN:181920
@SQ	

Step 2: Search for 'KI270728.1':
#### Homo sapiens chromosome 16 unlocalized genomic contig, GRCh38 reference primary assembly
GenBank: KI270728.1

    Source: https://www.ncbi.nlm.nih.gov/nuccore/KI270728.1
            
Step 3: Search for 'unlocalized genomic contig':  
        **Contig**:
    a contiguous sequence generated from determining the non-redundant path along an order set of component sequences. A contig should contain no gaps (Figure 1) but often the terms contig and scaffold are used interchangeably.   
    **Unlocalized Sequence**:  
    A sequence found in an assembly that is associated with a specific chromosome but cannot be ordered or oriented on that chromosome.   
    Source: https://www.ncbi.nlm.nih.gov/grc/help/definitions/
    
These unlocalized contigs represent the kind of sequences that you will find in the human genome indices in addition to the expected chromosome sequences.